ci/ray_ci/linux_container.py

```diff
@@ -0,0 +1,95 @@
+import os
+import subprocess
+import sys
+from typing import List, Optional
+
+from ci.ray_ci.container import Container
+
+_DOCKER_ENV = [
+    "BUILDKITE_BUILD_URL",
+    "BUILDKITE_BRANCH",
+    "BUILDKITE_COMMIT",
+    "BUILDKITE_JOB_ID",
+    "BUILDKITE_LABEL",
+    "BUILDKITE_PIPELINE_ID",
+]
+_DOCKER_CAP_ADD = [
+    "SYS_PTRACE",
+    "SYS_ADMIN",
+    "NET_ADMIN",
+]
+
+
+class LinuxContainer(Container):
+    def __init__(
+        self,
+        docker_tag: str,
+        volumes: Optional[List[str]] = None,
+        envs: Optional[List[str]] = None,
+    ) -> None:
+        super().__init__(docker_tag)
+        self.volumes = volumes or []
+        self.envs = envs or []
+        self.envs += _DOCKER_ENV
+
+    def install_ray(self, build_type: Optional[str] = None) -> List[str]:
+        env = os.environ.copy()
+        env["DOCKER_BUILDKIT"] = "1"
+        subprocess.check_call(
+            [
+                "docker",
+                "build",
+                "--pull",
+                "--build-arg",
+                f"BASE_IMAGE={self._get_docker_image()}",
```

Comment:
why is the base image and target tag having the same name? it is overwriting?

I did not notice this before but this feels quite confusing.

like the semantics now for windows and linux are different.

for windows, it is installing ray into a common / predefined base image.
but for linux, it is installing ray onto an existing image, and overwriting the tag.

Reply:
sound good, i can make windows accepting the base image with the same name format too, and retag the image before running the command